keps/sig-ui/5008-kep-template/README.md

```diff
@@ -0,0 +1,949 @@
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+One of Headlamp's strengths is that it was designed to both deliver a great
+end-user experience and be a platform for others to build upon and integrate
+with. This is demonstrated by the wide range of user stories it enables and one
+of the reasons we believe it can become a hub for the community.
+
+_**Note:** Using “Headlamp” in the following examples to avoid confusion._
+
+#### Story 1
+
+As an end user I want to get an overview of my cluster. I start the Headlamp
+Desktop app which picks up my clusters from my kubeconfig and I’m now able to
+explore my clusters.
+
+#### Story 2
+
+As a new Kubernetes user I want to learn more about Kubernetes. I use the
+Headlamp Desktop app to guide me in installing a local kubernetes cluster using
+minikube. Once my local cluster is installed I’m presented with a tutorial
+plugin which guides me through the various aspects of the UI and Kubernetes
+concepts.
+
+_**Note:** We have the minikube installer plugin in POC but the tutorial plugin
+is not something we currently have._
+
+#### Story 3
+
+As a cluster admin I was to provide a user interface for my users. I choose
+which plugins I want to include for my users and deploy Headlamp to my cluster.
+I have peace of mind knowing that Headlamp respects the RBAC rules that each
+user has.
+
+#### Story 4
+
+As a vendor I want to create a new Kubernetes user interface for my system. I
+take Headlamp Base and build a set of plugins and take some of the existing
+plugins to create my custom UI. I customize the UI with my company/product
+branding and provide that to my users with or without them knowing it’s
+Headlamp. If I desire, I can even allow my users to extend my customer UI using
+the same way I did.
+
+#### Story 5
+
+As a CNCF or external  project, I want to provide my users with a graphical
+experience but I don’t want to provide a full blown UI. I create a plugin that
+can be as simple as adding a column to a table (ex. OpenCost plugin) or as
+complex as creating a dedicated view with a sidebar entry and integrations
+across the various resource types (ex. Flux plugin). Example: See the OpenCost
+and Flux plugins.
+
+#### Story 6
+
+As a CNCF or external project I have a dedicated UI but that UI is not
+providing a full Kubernetes context. To provide my users with a Kubernetes
+context I create a link from my project to the Headlamp instance running in the
+cluster and create a plugin for Headlamp that provides the user in context
+views or links to my projects resources. This allows the user to easily
+navigate between the projects. (E.g. Backstage plugin)
+
+Out of the above stories, the current Kubernetes Dashboard only supports part
+of Story 3.
```

Comment:
It looks like stories 3-6 are based around plugin mechanism and can potentially be merged.

This is comparison that is not fully transparent. It shows use cases that Headlamp supports and completely skips stories that both projects support and those that only Kubernetes Dashboard support.

Reply:
I am updating this.